### PR TITLE
Delete all comments in one go

### PR DESCRIPTION
**Pull request by AhmedNadar**

### CODE REVIEW

**Comment by pirj** [comment]
Looks like a perfect TIL, short and useful. It takes a bit to understandhow it works though.

**Comment by thorncp** [comment]
What do you think about rephrasing this as learning about [`:g`][:g], and using deleting comments as an example?

[:g]: http://vimdoc.sourceforge.net/htmldoc/repeat.html#:g

**Comment by pirj** [comment]
By the way, til that `g/^#/d` does just the same

**Comment by JoelQ** [comment]
Thoughts on giving a quick rundown on how this works? Nothing in depth, just a few lines explaining what each section is doing. Otherwise this will look like a magical incantation for many reading this.

**Comment by AhmedNadar** [comment]
Here's my thought and how I understand this command.  


Command     | Details 
:----------:| --------
**:g**      | Gateway command, a prefix for commands don't have an assigned single-character command. Such as `gf` goto file, `gt` goto tab.
**:^#**     | Move to beginning of lines start with `#` character. 
**:normal** | Enter the Normal mode. Specify the mode.
**dd**      | Delete the whole lines
**:**       | By itself means enter the command-line mode so it run commands like `:tab`, `:wq`, ...etc.  



This is a gateway command `:g`, where the cursor moves to the beginning of lines `:` which starts with `#`, and in a normal mode `:normal` where I can do text manipulations; delete the whole lines `dd`.

@pirj works well, til a shorter version :+1:. I like the search `/` command.

**Comment by dcow** [comment]
What about multi-line string literals or `<<-`???

**Comment by phanimahesh** [comment]
@AhmedNadar Not exactly, `:g/pattern/command` applies command on all lines matching pattern.

so `:g/^#/d` applies `d` (delete) on all lines matching `^#` (starting with a hash)

The actual syntax is `:g<any-seperator><pattern><same-separator><command>` like most other similar vim commands (Eg:  `:s/pat/replacement/flags`, `:s=pat=replacement=flags`, and `:s/pat/replacement/flags` all do the same. choose a separator that minimises escapings in pattern and replacement.)

So `:g/` is no different from `:g:` or `:g=` for that matter.

**Comment by dcow** [comment]
I think you misunderstand. If your project contains a line that starts with a `#` that is not a comment, then bad things happen. I know how vim works.

**Comment by kchien** [comment]
I thought the `:` in `:g` meant a command for `ex`.
Then, the following `g` means to operate on all lines of the file.  You can make pattern substitutions or simply find lines matching a pattern.  (source: _Learning the vi and Vim editors_, by Arnold Robbins, Elbert Hannah and Linda Lamb.)

File: vim/delete_comments.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Delete all comments in one go
+
+Every Rails app you generate comes with many files such as routes that includes lots of comments.
+
+You can delete all these comments at once using the following vim command:
+
+```vimscript
+:g:^#:normal dd
```
Comment (tbodt) [inline vim/delete_comments.md:8]: It could be even shorter:

``` vimscript
:g/^#/d
```

For every line matching `^#`, run `d` (delete).

**Comment by phanimahesh** [comment]
@kchien Yes, the `:` in `:g` starts an ex-command.

`g` means operate on all lines matching the following pattern.
`v` would operate on lines not matching the given pattern.

**Comment by abhishekdgeek** [comment]
Nice!

**Comment by stefannibrasil** [comment]
This repo is no longer active and it's being archived. The contents will still be available as read-only. Thank you for your contributions!

**Pull request closed without merging**